icu4c/source/common/locid.cpp

```diff
@@ -477,6 +483,42 @@ Locale &Locale::operator=(const Locale &other)
     return *this;
 }
 
+Locale& Locale::operator=(Locale&& other) U_NOEXCEPT {
```

Comment:
I know it's not your code, but why does the copy assignment operator use uprv_malloc for fullName but uprv_strdup for baseName?  That looks really weird to me.

While we're here visiting this code, could you mind cleaning up the code for the copy assignment operator as well?  It will also make it easier to see the parallelism between copy assignment and move assignment.

Reply:
As far as I understand the code, the reason for handling `fullName` and `baseName` differently could have been some kind of desire to have one single point in the code where the `fullName` data is copied, the `uprv_strcpy()` call that's shared between cases where a new buffer is allocated with `uprv_malloc()` and cases where the object internal `fullNameBuffer` is used.

But I've now sent you PR #95 for review to rewrite the copy implementation along the lines of my proposed move implementation, the way I think the code becomes easier to follow and verify.